test/http2/test_http2_protocol.py

```diff
@@ -0,0 +1,216 @@
+
+import OpenSSL
+
+from netlib import http2
+from netlib import tcp
+from netlib import test
+from netlib.http2.frame import *
+from test import tutils
+
+
+class EchoHandler(tcp.BaseHandler):
+    sni = None
+
+    def handle(self):
+        v = self.rfile.readline()
+        self.wfile.write(v)
+        self.wfile.flush()
+
+
+class TestCheckALPNMatch(test.ServerTestBase):
+    handler = EchoHandler
+    ssl = dict(
+        alpn_select=http2.HTTP2Protocol.ALPN_PROTO_H2,
+    )
+
+    if OpenSSL._util.lib.Cryptography_HAS_ALPN:
+
+        def test_check_alpn(self):
+            c = tcp.TCPClient(("127.0.0.1", self.port))
+            c.connect()
+            c.convert_to_ssl(alpn_protos=[http2.HTTP2Protocol.ALPN_PROTO_H2])
+            protocol = http2.HTTP2Protocol(c)
+            assert protocol.check_alpn()
+
+
+class TestCheckALPNMismatch(test.ServerTestBase):
```

Comment:
But I have 2 different TCPServers with different SSL options.
How can I reinitialise that once the test class is setup?